### PR TITLE
app-crypt/tpm2-tss: add 4.0.0

### PR DESCRIPTION
**Pull request by salahcoronya**
New subslot for new major version. Also bumped to EAPI 8.

### CODE REVIEW

File: app-crypt/tpm2-tss/metadata.xml [hunk 1]
```diff
@@ -13,6 +13,8 @@
 		<flag name="fapi">Enable feature API (requires openssl as crypto backend)</flag>
 		<flag name="mbedtls">Use <pkg>net-libs/mbedtls</pkg> as crypto engine</flag>
 		<flag name="openssl">Use <pkg>dev-libs/openssl</pkg> as crypto engine</flag>
+		<flag name="policy">Enable policy library (requires openssl as crypto backend)</flag>
+
```
Comment (scardracs) [inline app-crypt/tpm2-tss/metadata.xml:17]: Drop the space :)

File: app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+inherit flag-o-matic linux-info multilib-minimal tmpfiles udev
+
+DESCRIPTION="TCG Trusted Platform Module 2.0 Software Stack"
+HOMEPAGE="https://github.com/tpm2-software/tpm2-tss"
+SRC_URI="https://github.com/tpm2-software/${PN}/releases/download/${PV}/${P}.tar.gz"
+
+LICENSE="BSD-2"
+SLOT="0/4"
+KEYWORDS="~amd64 ~arm ~arm64 ~loong ~ppc64 ~riscv ~x86"
+IUSE="doc +fapi +openssl mbedtls +policy static-libs test"
```
Comment (ceamac) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild:15]: Is `static-libs` still needed?

File: app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+		--with-udevrulesdir="$(get_udevdir)/rules.d" \
+		--with-udevrulesprefix=60- \
+		--with-sysusersdir="/usr/lib/sysusers.d" \
+		--with-tmpfilesdir="/usr/lib/tmpfiles.d"
+}
+
+multilib_src_install() {
+	default
+	keepdir /var/lib/tpm2-tss/system/keystore
+	rm -fr "${ED}/usr/$(get_libdir)/sysusers.d" || die
```
Comment (ceamac) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild:74]: Try not to use `rm -f`.  `rm -r` is enough.  `|| die` will trigger in the future if this is no longer needed; `rm -f` never dies.

**Review by ceamac — COMMENTED** [review]
Builds ok, tests run ok, `pkgcheck scan` is happy.

File: app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild (hunk 3 above)
Comment (salahcoronya) [reply app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild]: ok, dropped the -f

File: app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+		--with-udevrulesdir="$(get_udevdir)/rules.d" \
+		--with-udevrulesprefix=60- \
+		--with-sysusersdir="/usr/lib/sysusers.d" \
+		--with-tmpfilesdir="/usr/lib/tmpfiles.d"
+}
+
+multilib_src_install() {
+	default
+	keepdir /var/lib/tpm2-tss/system/keystore
+	rm -r "${ED}/usr/$(get_libdir)/sysusers.d" || die
```
Comment (ceamac) [inline app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild:74]: Now this rm fails.  It was also failing before, but -f was hiding the error.

`--with-sysusersdir="/usr/lib/sysusers.d"` is used in configure phase, but `/usr/$(get_libdir)/sysusers.d` in install.  On amd64 this translates to `/usr/lib64/sysusers.d`, which does not exist.

**Review by ceamac — CHANGES_REQUESTED** [review]
Please also check if `sys-apps/util-linux` is required in RDEPEND (probably because of `libuuid.so.1`)

Thank you!

File: app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild (hunk 4 above)
Comment (salahcoronya) [reply app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild]: I edited the config options and patches the Makefile to remove the need for the rm statements. sys-apps/util-linux is part of the system set, do I really need it in RDEPEND?

Comment (ceamac) [reply app-crypt/tpm2-tss/tpm2-tss-4.0.0.ebuild]: Good point.  I'm not sure about that, I'll ask.

**Review by ceamac — APPROVED** [review]

**Pull request closed without merging**